flowr-compilers/src/main/java/com/fueled/flowr/compilers/DeepLinkAnnotationCompiler.java

```diff
@@ -0,0 +1,306 @@
+                .addTypeVariable(fragmentTypeVariableName)
+                .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+                .addSuperinterface(ClassName.get(FLOWR_INTERNAL_PACKAGE_NAME, "FlowrDeepLinkHandler"));
+    }
+
+    /**
+     * Create the bundleUriInfo method of FlowrDeepLinkHandlerImpl class. This method extract the variables from a URL and add them to a Bundle.
+     * The structure is <variable, value>.
+     * For example the pattern in the DeepLink annotation \"/user/{userId}\", if the user access the app through \"/user/1234\", the function will return <userId,1234>.
+     * Please note that all the variable will a string.
+     *
+     * @return a Bundle that contains the variables from the deep link.
+     */
+    private static MethodSpec bundleUriInfo() {
+        return MethodSpec.methodBuilder("bundleUriInfo")
+                .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                .returns(bundleClassName)
+                .addParameter(uriClassName, "uri")
+                .addParameter(String.class, "pattern")
+                .addCode("        String regex = getRegexPattern(pattern);\n" +
+                        "        String path = uri.getPath();\n" +
+                        "        $T m = Pattern.compile(regex).matcher(path);\n" +
+                        "        if (m.matches()) {\n" +
+                        "            $T data = new Bundle();\n" +
+                        "            data.putString($T.DEEP_LINK_URL, uri.toString());\n" +
+                        "            $T<String> params = getNamedGroupCandidates(regex).iterator();\n" +
+                        "            //start at 1 because 0 is the searched string\n" +
+                        "            int i = 1;\n" +
+                        "            while (params.hasNext()) {\n" +
+                        "                String bip = params.next();\n" +
+                        "                data.putString(bip, m.group(i));\n" +
+                        "                i++;\n" +
+                        "            }\n" +
+                        "            return data;\n" +
+                        "        }\n" +
+                        "        return null;", Matcher.class, bundleClassName, ClassName.get(FLOWR_PACKAGE_NAME, "Flowr"), Iterator.class)
+                .build();
+
+    }
+
+    /**
+     * Create the getRegexPattern method that generate the regular expression to extract variables form a URL.
+     *
+     * @return RegExp to extract variables form a URL.
+     */
+    private static MethodSpec getRegexPattern() {
+        return MethodSpec.methodBuilder("getRegexPattern")
+                .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                .returns(String.class)
+                .addParameter(String.class, "pattern")
+                .addStatement("return pattern.replaceAll(\"\\\\{(.+?)\\\\}\", \"(?<$L>\\\\.+?)\")", "$1")
```

Comment:
@chetan-fueled no it would get to confusion. I?f we abstracted the generated code out of the generating, it will be extremely hard to read.